### ISSUE TITLE
IOS build error

### ISSUE THREAD

**Issue by lexyfeito** [jonasbark/flutter_stripe_payment#131]
Error when building for IOS

`error: incompatible block pointer types sending '__strong STPAPIResponseBlock _Nonnull' (aka 'void (^__strong)(ResponseType  _Nullable __strong, NSHTTPURLResponse * _Nullable __strong, NSError * _Nullable __strong)') to parameter of type 'void (^ _Nonnull)(STPSource * _Nullable __strong, NSHTTPURLResponse * _Nullable __strong, NSError * _Nullable __strong)'
                                                 completion:completion];
                                                            ^~~~~~~~~~
    In file included from /Users/lexyfeito/AndroidStudioProjects/nanu_cakes/ios/Pods/Stripe/Stripe/STPAPIClient.m:16:
    In file included from /Users/lexyfeito/AndroidStudioProjects/nanu_cakes/ios/Pods/Stripe/Stripe/STPAPIClient+Private.h:12:
    /Users/lexyfeito/AndroidStudioProjects/nanu_cakes/ios/Pods/Stripe/Stripe/STPAPIRequest.h:50:65: note: passing argument to parameter 'completion' here
                                    completion:(STPAPIResponseBlock)completion;`

flutter doctor

`Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
[✓] Android Studio (version 3.5)
[✓] Connected device (1 available)

• No issues found!`

**Comment by o-gi** [comment]
I got same error after Xcode update.
Probably this is a reason.

https://stackoverflow.com/questions/60841771/stripe-ios-sdk-incompatible-block-pointer-types-sending

**Comment by herrmartell** [comment]
I'm having the same issue. 

I see that this plugin is requiring Stripe version 18.0.0. AFAIK while researching on SO, this was solved on Stripe v 19.0.1

Maybe the author could give that a check?

**Comment by anthonycmeow** [comment]
Same issue here, can't build. @jonasbark could you please take a look?

**Comment by eleanoryy** [comment]
same issue here.
After updating XCode 11.4, I can not build app depending on Stripe v18.4.0
For building app, it seems to require Stripe v19.0.1

**Comment by jonasbark** [comment]
Feel free to create a Pull request for this. 
I may be able to look at it tomorrow or the day after.

**Comment by jonasbark** [comment]
Please have a go with the current dev branch:
https://github.com/jonasbark/flutter_stripe_payment/tree/dev

https://dart.dev/tools/pub/dependencies#git-packages

If it works for you and Xcode 11.4, I'll merge that branch and create a new version.

**Comment by AlexSze** [comment]
/Users/alex/Developer/flutter/.pub-cache/git/flutter_stripe_payment-d0a5b4736a8f4267d6211734f3f58e87dc9c4f1c/ios/Classes/TPSStripeManager.m:309:42: warning: comparison of distinct pointer types ('NSString *' and 'NSNull * _Nonnull') [-Wcompare-distinct-pointer-types]
        if (_stripeAccount && _stripeAccount != [NSNull null]) {
                              ~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~
    /Users/alex/Developer/flutter/.pub-cache/git/flutter_stripe_payment-d0a5b4736a8f4267d6211734f3f58e87dc9c4f1c/ios/Classes/TPSStripeManager.m:731:42: warning: 'startPollingSourceWithId:clientSecret:timeout:completion:' is deprecated: You should poll your own backend to update based on source status change webhook events it may receive. [-Wdeprecated-declarations]
                            [stripeAPIClient startPollingSourceWithId:sourceID clientSecret:clientSecret timeout:10 completion:^(STPSource *source, NSError *error) {
                                             ^
    In module 'Stripe' imported from /Users/alex/Developer/flutter/.pub-cache/git/flutter_stripe_payment-d0a5b4736a8f4267d6211734f3f58e87dc9c4f1c/ios/Classes/TPSStripeManager.h:11:
    /Users/alex/Desktop/picare-user/flutter_app/ios/Pods/Stripe/Stripe/PublicHeaders/STPAPIClient.h:352:243: note: 'startPollingSourceWithId:clientSecret:timeout:completion:' has been explicitly marked deprecated here
    - (void)startPollingSourceWithId:(NSString *)identifier clientSecret:(NSString *)secret timeout:(NSTimeInterval)timeout completion:(STPSourceCompletionBlock)completion NS_EXTENSION_UNAVAILABLE("Source polling is not available in extensions") DEPRECATED_MSG_ATTRIBUTE("You should poll your own backend to update based on source status change webhook events it may receive.");
                                                                                                                                                                                                                                                      ^
    In module 'Foundation' imported from /Users/alex/Developer/flutter/.pub-cache/git/flutter_stripe_payment-d0a5b4736a8f4267d6211734f3f58e87dc9c4f1c/ios/Classes/TPSStripeManager.h:9:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk/usr/include/AvailabilityMacros.h:182:64: note: expanded from macro 'DEPRECATED_MSG_ATTRIBUTE'
                #define DEPRECATED_MSG_ATTRIBUTE(s) __attribute__((deprecated(s)))
                                                                   ^
    /Users/alex/Developer/flutter/.pub-cache/git/flutter_stripe_payment-d0a5b4736a8f4267d6211734f3f58e87dc9c4f1c/ios/Classes/TPSStripeManager.m:798:20: warning: 'setPublishableKey:' is deprecated: If you used [STPPaymentConfiguration sharedConfiguration].publishableKey, use [STPAPIClient sharedClient].publishableKey instead. If you passed a STPPaymentConfiguration instance to an SDK component, create an STPAPIClient, set publishableKey on it, and set the SDK component's APIClient property. [-Wdeprecated-declarations]
        [configuration setPublishableKey:nextPublishableKey];
                       ^
    In module 'Stripe' imported from /Users/alex/Developer/flutter/.pub-cache/git/flutter_stripe_payment-d0a5b4736a8f4267d6211734f3f58e87dc9c4f1c/ios/Classes/TPSStripeManager.h:11:
    /Users/alex/Desktop/picare-user/flutter_app/ios/Pods/Stripe/Stripe/PublicHeaders/STPPaymentConfiguration.h:126:50: note: property 'publishableKey' is declared deprecated here
    @property (nonatomic, copy, readwrite) NSString *publishableKey DEPRECATED_MSG_ATTRIBUTE("If you used [STPPaymentConfiguration sharedConfiguration].publishableKey, use [STPAPIClient sharedClient].publishableKey instead. If you passed a STPPaymentConfiguration instance to an SDK component, create an STPAPIClient, set publishableKey on it, and set the SDK component's APIClient property.");
                                                     ^
    /Users/alex/Desktop/picare-user/flutter_app/ios/Pods/Stripe/Stripe/PublicHeaders/STPPaymentConfiguration.h:126:65: note: 'setPublishableKey:' has been explicitly marked deprecated here
    @property (nonatomic, copy, readwrite) NSString *publishableKey DEPRECATED_MSG_ATTRIBUTE("If you used [STPPaymentConfiguration sharedConfiguration].publishableKey, use [STPAPIClient sharedClient].publishableKey instead. If you passed a STPPaymentConfiguration instance to an SDK component, create an STPAPIClient, set publishableKey on it, and set the SDK component's APIClient property.");
                                                                    ^
    In module 'Foundation' imported from /Users/alex/Developer/flutter/.pub-cache/git/flutter_stripe_payment-d0a5b4736a8f4267d6211734f3f58e87dc9c4f1c/ios/Classes/TPSStripeManager.h:9:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk/usr/include/AvailabilityMacros.h:182:64: note: expanded from macro 'DEPRECATED_MSG_ATTRIBUTE'
                #define DEPRECATED_MSG_ATTRIBUTE(s) __attribute__((deprecated(s)))
                                                                   ^
    /Users/alex/Developer/flutter/.pub-cache/git/flutter_stripe_payment-d0a5b4736a8f4267d6211734f3f58e87dc9c4f1c/ios/Classes/TPSStripeManager.m:1728:16: warning: 'STPBillingAddressFieldsZip' is deprecated: Use STPBillingAddressFieldsPostalCode [-Wdeprecated-declarations]
            return STPBillingAddressFieldsZip;
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~
                   STPBillingAddressFieldsPostalCode
    In module 'Stripe' imported from /Users/alex/Developer/flutter/.pub-cache/git/flutter_stripe_payment-d0a5b4736a8f4267d6211734f3f58e87dc9c4f1c/ios/Classes/TPSStripeManager.h:11:
    /Users/alex/Desktop/picare-user/flutter_app/ios/Pods/Stripe/Stripe/PublicHeaders/STPAddress.h:48:47: note: 'STPBillingAddressFieldsZip' has been explicitly marked deprecated here
        STPBillingAddressFieldsZip __attribute__((deprecated("Use STPBillingAddressFieldsPostalCode", "STPBillingAddressFieldsPostalCode"))) = STPBillingAddressFieldsPostalCode,
                                                  ^






it didn't work for us

**Comment by jonasbark** [comment]
@AlexSze those are warnings. Do you get any errors?

**Comment by AlexSze** [comment]
actually i got build failed

**Comment by jonasbark** [comment]
I don't think this is related to stripe_payment:
```
ld: framework not found FirebaseInstanceID
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by AlexSze** [comment]
Ok, thanks, I will do some more research on it

**Comment by jonasbark** [comment]
@eleanoryy @lexyfeito  @o-gi @herrmartell @anthonycmeow what about you? does the dev branch work for you?

**Comment by herrmartell** [comment]
I strongly believe the dev branch fixes the issue. I cannot guarantee it 100% because after that, I'm getting this other error

> Failed to find assets path for "Frameworks/App.framework/flutter_assets"
[VERBOSE-2:engine.cc(127)] Engine run configuration was invalid.
[VERBOSE-2:shell.cc(437)] Could not launch engine with configuration.

But I don't think this is related to this plugin, is it?

**Comment by jonasbark** [comment]
No, that's highly unlikely.

**Comment by pradt** [comment]
I'm receiving this after upgrading flutter.

**Comment by pradt** [comment]
> I got same error after update XCode 11.4.
> Probably this is a reason.
> 
> https://stackoverflow.com/questions/60841771/stripe-ios-sdk-incompatible-block-pointer-types-sending

Hi I'm posting this to state that I've resolved this issue based on this stackoverflow. What worked for me was the answer stating that they had to manually update podfile.lock file to update the Stripe version to 19.0.1. 

My Podfile.lock entry for Stripe now reads :
`- Stripe (19.0.1)
`
After which I ran the following in the terminal :
`pod update Stripe` 

I then rebuilt my app and everything seems to be working, at least getting through the build process. 

Please note, that the marked answer of deleting the podfile.lock didn't work for me.

**Comment by jonasbark** [comment]
@pradt can you try my solution I posted few days before? 
https://github.com/jonasbark/flutter_stripe_payment/issues/131#issuecomment-606716054

**Comment by herrmartell** [comment]
@jonasbark I can confirm it works

**Comment by jonasbark** [comment]
Fixed with 1.0.7

**Comment by wilpar** [comment]
I'm now getting this error, and am on 1.0.8.  I see Stripe pod is now on 21.0.1.  I've tried specifying the pod in the podfile directly, 19.0.1, and 21.0.1, same error both times - did not find header Stripe.h
